applications/StructuralMechanicsApplication/custom_strategies/custom_builder_and_solver/explicit_builder_and_solver.hpp

```diff
@@ -0,0 +1,461 @@
+        #endif
+
+        vector<unsigned int> node_partition;
+        OpenMPUtils::CreatePartition(number_of_threads, pNodes.size(), node_partition);
+
+        vector<unsigned int> element_partition;
+        OpenMPUtils::CreatePartition(number_of_threads, pElements.size(), element_partition);
+
+
+        #pragma omp parallel
+        {
+
+            #pragma omp for
+
+                for(int k=0; k<number_of_threads; k++)
+                {
+                    typename NodesArrayType::iterator i_begin=pNodes.ptr_begin()+node_partition[k];
+                    typename NodesArrayType::iterator i_end=pNodes.ptr_begin()+node_partition[k+1];
+
+                    for(ModelPart::NodeIterator i=i_begin; i!= i_end; ++i)
+                    {
+                        double& nodal_mass    =  i->FastGetSolutionStepValue(NODAL_MASS);
+                        nodal_mass = 0.0;
+
+                        if (i->HasDofFor(ROTATION_X))
+                        {
+                            array_1d<double,3>& nodal_inertia = i->FastGetSolutionStepValue(NODAL_INERTIA);
+                            noalias(nodal_inertia) = ZeroVector(3);
+                        }
+
+                    }
+                }
+        }
+    
+        #pragma omp parallel
+        {
+            int k = OpenMPUtils::ThisThread();
+            typename ElementsArrayType::iterator ElemBegin = pElements.begin() + element_partition[k];
+            typename ElementsArrayType::iterator ElemEnd = pElements.begin() + element_partition[k + 1];
+
+            for (typename ElementsArrayType::iterator itElem = ElemBegin; itElem != ElemEnd; itElem++)  
+            {
+                //Getting nodal mass and inertia from element
+                Vector Testtemp;
+
+                // this function needs to be implemented in the respective 
+                // element to provide inertias and nodal masses 
+                itElem->AddExplicitContribution(Testtemp,RESIDUAL_VECTOR,NODAL_INERTIA,rCurrentProcessInfo);
+            }
+        }
+        
+        KRATOS_CATCH( "" )
+    }
+
+    //**************************************************************************
+    //**************************************************************************
+
+    void BuildRHS(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemVectorType& b)
+    {
+        KRATOS_TRY
+        
+        // Compute condition contributions to RHS.
+        CalculateAndAddConditionsRHS(pScheme, rModelPart);
+
+        // Compute element contributions to RHS.
+        CalculateAndAddElementsRHS(pScheme, rModelPart);
+
+        
+        KRATOS_CATCH( "" )
+
+    }
+
+
+    
+    
+    //***************************************************************************
+    //***************************************************************************
+
+   
+    void CalculateAndAddConditionsRHS(typename TSchemeType::Pointer pScheme, ModelPart& rModelPart )
```

Comment:
this shoulb be a templated function that acceps both elements or conditions, see [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MappingApplication/custom_utilities/mapper_utilities.h#L149)

Reply:
this has neither an element or a condition as input. Can you explain what you mean?